test/helper/identityMock.ts

```diff
@@ -0,0 +1,40 @@
+import { MockAgent, setGlobalDispatcher } from 'undici'
```

Comment:
mmm?? do we need to use this? If we are intercepting and etc, then stub maybe?

However, can see quite a few use cases of this, myself I used nock so would be great to get Matts opinion on this since we discussed something about builtin `fetch`

Reply:
`nock` doesn't work with node 18 fetch

Reply:
there will be more elaborate intercepting when we get to testing `accept_a` or `accept_b` first flows